### PR TITLE
Mes 1567 accessibility needs - bigger typography

### PR DESCRIPTION
**Pull request by buzzub**
## Description and relevant Jira numbers

- Added a Cordova plugin to get the text zoom level from the iOS Accessibility settings (up to 144%).
- Created new CSS classes of `text-zoom-regular`, `text-zoom-large` and `text-zoom-x-large` that apply to the app root based on the zoom level.
- Updated typography file to include larger font sizes for these classes.
- Updated various components and SCSS files to improve consistency (e.g. changing some divs to be h3, changing margins and spacing to work well all three zoom levels).

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [ ] Code has been tested manually
- [ ] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts [hunk 1]
```diff
@@ -27,35 +27,35 @@ describe('VehicleDetailsComponent', () => {
...
+    // it('should not show number of seats when showNumberOfSeats is false', () => {
+    //   component.showDimensions = false;
+    //   component.showNumberOfSeats = false;
+    //   fixture.detectChanges();
+    //   const ionCols = fixture.debugElement.queryAll(By.css('ion-col'));
+    //   expect(ionCols.length).toBe(0);
+    // });
   });
 
 });
```
Comment (rossfauldsbjss) [inline src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts]: Commented out tests?  why?

File: src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts [hunk 2]
```diff
@@ -27,35 +27,35 @@ describe('VehicleDetailsComponent', () => {
...
-    });
-
-    it('should not show number of seats when showNumberOfSeats is false', () => {
-      component.showDimensions = false;
-      component.showNumberOfSeats = false;
-      fixture.detectChanges();
-      const ionCols = fixture.debugElement.queryAll(By.css('ion-col'));
-      expect(ionCols.length).toBe(0);
-    });
+    // it('should show vehicle details when showDimensions is true', () => {
```
Comment (szabolcs-keresztes) [inline src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts]: Did you forget about these?

File: src/pages/journal/components/vehicle-details/vehicle-details.scss [hunk 3]
```diff
@@ -1,9 +1,11 @@
 vehicle-details {
-  .vehicle-details-grid {
-    font-size: 20px;
-    padding: 0 0.6rem;
-  }
+  // .vehicle-details-grid {
```
Comment (szabolcs-keresztes) [inline src/pages/journal/components/vehicle-details/vehicle-details.scss]: Can be removed

File: src/app/app.component.ts [hunk 4]
```diff
@@ -33,4 +38,39 @@ export class App {
   loadAppInfo() {
     this.store$.dispatch(new LoadAppInfo());
   }
+
+  configureAccessibility() {
+    if (this.platform.is('ios') && window && window.MobileAccessibility) {
+      window.MobileAccessibility.getTextZoom(this.getTextZoomCallback);
```
Comment (ammarhaiderbjss) [inline src/app/app.component.ts]: Is this the correct usage of the plugin? Could we not use the ionic native wrapper?

```
import { MobileAccessibility } from '@ionic-native/mobile-accessibility';


constructor(private mobileAccessibility: MobileAccessibility) { }

...

if(this.mobileAccessibility. getTextZoom(this.getTextZoomCallback);
```

Comment (ammarhaiderbjss) [reply src/app/app.component.ts]: Appending it to the global `window` object makes it invisible to Angular's DI system too

Comment (buzzub) [reply src/app/app.component.ts]: That's what the docs say, but in reality it doesn't work.

File: src/pages/journal/components/indicators/indicators.scss [hunk 5]
```diff
@@ -4,6 +4,14 @@ indicators {
     img {
       width: 29px;
       height: 25px;
+      .text-zoom-large & {
```
Comment (ammarhaiderbjss) [inline src/pages/journal/components/indicators/indicators.scss]: Is the idea that we configure these CSS classes individually for each of our UI components?

Comment (buzzub) [reply src/pages/journal/components/indicators/indicators.scss]: The typography should scale fine without changes, but there might be times when you want more control (like making images match the text sizes) and this approach would work. Possibly not ideal. Ideally we would use rem/em to scale from a base font, but the base font is always fixed in this approach.

Comment (buzzub) [reply src/pages/journal/components/indicators/indicators.scss]: And I used the `.text-zoom-large &` ampersand syntax as it allows the extra classes to stay near the element being overridden. Just seems neater.

File: src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts (hunk 1 above)
Comment (buzzub) [reply src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts]: The vehicle details component no longer includes those elements and so yes, the tests should be removed actually

**Pull request merged**